bundles/CoreBundle/DependencyInjection/Compiler/MarshallerLocatorPass.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Enterprise License (PEL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PEL
+ */
+declare(strict_types=1);
+
+namespace Pimcore\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use Pimcore\Bundle\DataHubBundle\GraphQL\Service;
+use Pimcore\Element\MarshallerService;
+use Symfony\Component\Config\Definition\Exception\InvalidDefinitionException;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+
+
+/**
+ * @internal
+ */
+final class MarshallerLocatorPass implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->processFieldDefinitionMarshallers($container);
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     */
+    protected function processFieldDefinitionMarshallers(ContainerBuilder $container)
+    {
+        $serviceDefinition = $container->getDefinition(MarshallerService::class);
+         $this->buildSupportedFielddefinitionMarshaller(
+            'query',
+            $container,
+            $serviceDefinition,
+            'field definition marshal',
+            'pimcore.dataobject.fielddefinition.marshaller'
+        );
+    }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param Definition $definition
+     * @param string $type
+     * @param string $tag
+     * @param string $argument
+     */
+    private function createLocatorForTaggedServices(
```

Comment:
I think the pass can be replaced by `!tagged_locator`, see: 
https://symfony.com/doc/current/service_container/service_subscribers_locators.html#indexing-the-collection-of-services

Reply:
Would be a nice approach, but not supported in symfony 3.4

https://olvlvl.com/2019-09-symfony-tagged-service-locator

Reply:
Aaah, ok, then we should mark it already as internal and change it in X 😊